src/dodrio_berry_picking.c

```diff
@@ -1,870 +1,2437 @@
+    count = gUnknown_203F3E0->unk24;
+    for (i = 0; i < count; i++)
+        sub_8153DA8(i, 4);
+}
+
+static void sub_815293C(void)
 {
-    .bg = 0,
-    .tilemapLeft = 5,
-    .tilemapTop = 8,
-    .width = 19,
-    .height = 3,
-    .paletteNum = 13,
-    .baseBlock = 0x13,
-};
+    sub_81527D0();
+    if (gUnknown_203F3E0->unk40 > 9)
+        sub_8152910();
+    else
+        sub_81528D0();
+
+    sub_8153FC8(gUnknown_203F3E0->unk40);
+}
 
-/*static*/ const u8 sUnknown_8471F50[5][5][11] =
+// This function is literally the same as the one above...Why?
+static void sub_8152970(void)
 {
-    {
-        {0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 0},
-    },
-    {
-        {0, 1, 2, 3, 4, 5, 6, 3, 8, 9, 0},
-        {0, 1, 2, 5, 6, 3, 4, 5, 8, 9, 0},
-    },
-    {
-        {0, 1, 2, 3, 4, 5, 6, 7, 2, 9, 0},
-        {0, 1, 4, 5, 6, 7, 2, 3, 4, 9, 0},
-        {0, 1, 6, 7, 2, 3, 4, 5, 6, 9, 0},
-    },
-    {
-        {0, 1, 2, 3, 4, 5, 6, 7, 8, 1, 0},
-        {0, 3, 4, 5, 6, 7, 8, 1, 2, 3, 0},
-        {0, 5, 6, 7, 8, 1, 2, 3, 4, 5, 0},
-        {0, 7, 8, 1, 2, 3, 4, 5, 6, 7, 0},
-    },
-    {
-        {0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 0},
-        {2, 3, 4, 5, 6, 7, 8, 9, 0, 1, 2},
-        {4, 5, 6, 7, 8, 9, 0, 1, 2, 3, 4},
-        {6, 7, 8, 9, 0, 1, 2, 3, 4, 5, 6},
-        {8, 9, 0, 1, 2, 3, 4, 5, 6, 7, 8},
-    },
-};
+    sub_81527D0();
+    if (gUnknown_203F3E0->unk40 > 9)
+        sub_8152910();
+    else
+        sub_81528D0();
 
-/*static*/ const u8 sUnknown_8472063[5][5][3] =
+    sub_8153FC8(gUnknown_203F3E0->unk40);
+}
+
+static void sub_81529A4(u8 arg0, u8 *arg1, u8 *arg2)
 {
+    switch (arg0)
     {
-        {4, 5, 6},
-    },
-    {
-        {3, 4, 5},
-        {5, 6, 3},
-    },
-    {
-        {4, 5, 6},
-        {6, 7, 2},
-        {2, 3, 4},
-    },
-    {
-        {3, 4, 5},
-        {5, 6, 7},
-        {7, 8, 1},
-        {1, 2, 3},
-    },
-    {
-        {4, 5, 6},
-        {6, 7, 8},
-        {8, 9, 0},
-        {0, 1, 2},
-        {2, 3, 4},
-    },
-};
+    case 1:
+        *arg1 = 4, *arg2 = 7;
+        break;
+    case 2:
+        *arg1 = 3, *arg2 = 8;
+        break;
+    case 3:
+        *arg1 = 2, *arg2 = 9;
+        break;
+    case 4:
+        *arg1 = 1, *arg2 = 10;
+        break;
+    case 5:
+        *arg1 = 0, *arg2 = 11;
+        break;
+    }
+}
 
-/*static*/ const u8 sUnknown_84720AE[5][5][3] =
+static bool32 sub_8152A00(void)
 {
+    u8 i, count;
+
+    count = gUnknown_203F3E0->unk24;
+    for (i = 1; i < count; i++)
     {
-        {1, 0, 1},
-    },
-    {
-        {1, 0, 1},
-        {0, 1, 0},
-    },
-    {
-        {2, 0, 1},
-        {0, 1, 2},
-        {1, 2, 0},
-    },
-    {
-        {3, 0, 1},
-        {0, 1, 2},
-        {1, 2, 3},
-        {2, 3, 0},
-    },
+        if (gUnknown_203F3E0->unk158[i] == 0)
+            gUnknown_203F3E0->unk158[i] = sub_815A5E8(i);
+    }
+
+    // This loop won't ever run, the seemingly poitnless assingment below is to make the compiler
```

Comment:
typo

Reply:
lol, that typo is in emerald too. blame egg